[Features] Allow setting legend title alignment (#23140)
matplotlib/matplotlib: doc/users/next_whats_new/legend_align.rst

Legend can control alignment of title and handles
-------------------------------------------------

> Legend now supports control the alignment of title and handles using the
> keyword argument ``alignment``. User can also use `.set_alignment` method
> on the `Legend` class to control the alignment.

Reviewer: ```suggestion
`.Legend` now supports control the alignment of title and handles via the
keyword argument ``alignment``. You can also use `.Legend.set_alignment`
to control the alignment on existing Legends.
```

Slight rewording and fixes the linking issue in doc build.
